### ISSUE TITLE
Demo

### ISSUE THREAD

**Issue by DawChihLiou** [tantaraio/voy#15]

**Comment by josephrocca** [comment]
I've just tested the code in the readme on node.js and I'm getting a bunch of different import errors. I've also tried to get it working in the browser but am getting this:
```
Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "application/wasm". Strict MIME type checking is enforced for module scripts per HTML spec.
```
I'm really only interested in getting it working in the browser - just tested in node.js after getting this wasm module loading error. Seems like it's only phase 2: https://github.com/WebAssembly/esm-integration/tree/main/proposals/esm-integration ?

**Comment by DawChihLiou** [comment]
Hey @josephrocca thanks for your question! Please check out https://github.com/tantaraio/voy/issues/6#issuecomment-1490426936 for more info.

**Comment by josephrocca** [comment]
Do you not intend to make it import-able with a simple `import` statement in the browser? Or is this just a temporary workaround?

**Comment by DawChihLiou** [comment]
Currently WASM modules requires dynamic import in browsers and node.js. I'll look into it to make it easier to import in browsers. Do you mind sharing your use case with me so I understand more about your needs? Thanks!

**Comment by josephrocca** [comment]
Use case: https://github.com/josephrocca/OpenCharacters - to handle character memory storage/recall

Currently I'm doing a brute force cosine similarity search - it'll be fine for short chats but might become slow if the character/agent accumulates thousands of memories.

**Comment by GarciaLnk** [comment]
The simplest solution to support browser imports would be building and bundling a version targeting browsers (as I suggested in #6) inside the current npm package and doing `import * as voy from "voy-search/web"` when needed.

It's not a perfect solution, but it's better than the current options (my current solution is having a git submodule linked and running `git -C voy pull && wasm-pack build -t web voy` on pnpm preinstall).

**Comment by MentalGear** [comment]
@GarciaLnk sounds neat, would you be able to add a PR for this ?

**Comment by DawChihLiou** [comment]
@GarciaLnk have you tried https://github.com/tantaraio/voy/issues/6#issuecomment-1490426936? If it doesn't work, I'll work on the build for web.

```
<script src="https://www.unpkg.com/voy-search@0.6.1/voy_search.js" type="module"></script>
```

**Comment by DawChihLiou** [comment]
Hey @kwkr thanks for the comment! The configuration besides the "main" property in your package.json should work. I'm more than happy to review if you'd like to creat a PR! Thanks so much!

**Comment by kwkr** [comment]
I wasn't precise with specifying which `package.json` I modified. I needed to modify the `package.json` of the `voy-search` inside `node_modules`. Should this work with Node without those modifications?

**Comment by DawChihLiou** [comment]
Yes that's correct. If it doesn't work, please let me know. I'm currently on vacation but I'll take a look at it as soon as I get back. Thanks!

**Comment by DawChihLiou** [comment]
Hey @kwkr just came across your [integration in LangChain.js](https://js.langchain.com/docs/modules/data_connection/vectorstores/integrations/voy). Amazing work! Is your issue resolved?

**Comment by kwkr** [comment]
Yes, I managed to integrate it. The question is whether the information I needed in order to do that should be included in the README of your package as well because as I mentioned, I needed a few workarounds.

**Comment by DawChihLiou** [comment]
@kwkr I'll work in a build for Node. Thanks for your suggestion!
